packages/playwright-core/src/client/browserContext.ts

```diff
@@ -145,19 +145,26 @@ export class BrowserContext extends ChannelOwner<channels.BrowserContextChannel>
 
   _onRoute(route: network.Route, request: network.Request) {
     for (const routeHandler of this._routes) {
-      if (routeHandler.matches(request.url())) {
+      if (!routeHandler.matches(request.url()))
+        continue;
+      // Immediately deactivate based on |times|.
+      if (routeHandler.willExpire())
```

Comment:
Are you sure it works fine with the for/of iteration?

Reply:
1) yes 2) we return after a single iteration.